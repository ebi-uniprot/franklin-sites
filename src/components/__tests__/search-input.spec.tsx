import { screen, render, fireEvent } from '@testing-library/react';

import SearchInput from '../search-input';

describe('SearchInput', () => {
  it('should render', () => {
    const { asFragment } = render(
      <SearchInput placeholder="This is the initial value" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should focus on text input when suffix is clicked', () => {
    render(<SearchInput placeholder="This is the initial value" />);
    const suffix = screen.getByTestId('search-input-suffix');
    fireEvent.click(suffix);
    const input = screen.queryByTestId('search-input');
    expect(document.activeElement).toEqual(input);
  });

  it('should should not call onChange when no value is present and suffix is clicked', () => {
    const onChange = jest.fn();
    render(<SearchInput onChange={onChange} />);
    const suffix = screen.getByTestId('search-input-suffix');
    fireEvent.click(suffix);
    expect(onChange).not.toHaveBeenCalled();
  });

  it('should call onChange with "" when value is present and suffix is clicked', () => {
    const onChange = jest.fn();
    render(<SearchInput onChange={onChange} value="foo" />);
    const suffix = screen.getByTestId('search-input-suffix');
    fireEvent.click(suffix);
    expect(onChange).toHaveBeenCalled();
  });

  it('should not call onChange with "" when value is present and suffix is clicked and isLoading', () => {
    const onChange = jest.fn();
    render(<SearchInput onChange={onChange} value="foo" isLoading />);
    const suffix = screen.getByTestId('search-input-suffix');
    fireEvent.click(suffix);
    expect(onChange).not.toHaveBeenCalledWith({ target: { value: '' } });
  });
});
