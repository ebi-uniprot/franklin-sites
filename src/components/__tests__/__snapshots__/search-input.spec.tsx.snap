// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should render 1`] = `
<DocumentFragment>
  <span
    class="search-input"
  >
    <input
      data-testid="search-input"
      placeholder="This is the initial value"
      type="text"
      value=""
    />
    <span
      class="search-input__suffix"
      data-testid="search-input-suffix"
      role="presentation"
    >
      <test-file-stub
        height="14"
        width="14"
      />
    </span>
  </span>
</DocumentFragment>
`;
